### PR TITLE
New spack package nvtop

### PR DESCRIPTION
**Pull request by perrymil**

### CODE REVIEW

File: var/spack/repos/builtin/packages/nvtop/package.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# them, you can save this file and test your package like this:
+#
+#     spack install nvtop
+#
+# You can edit this file again by typing:
+#
+#     spack edit nvtop
+#
+# See the Spack documentation for more information on packaging.
+# ----------------------------------------------------------------------------
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/nvtop/package.py]: please remove FIXME

File: var/spack/repos/builtin/packages/nvtop/package.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    url      = "https://github.com/Syllo/nvtop/archive/1.1.0.tar.gz"
+
+    version('1.1.0', sha256='00470cde8fc48d5a5ed7c96402607e474414d94b562b21189bdde1dbe6b1d1f3')
+
+    depends_on('cmake')
+    depends_on('ncurses')
+    depends_on('git')
+    depends_on('cuda')
+    depends_on('py-pynvml', type=('build', 'run'))
+    depends_on('py-nvidia-ml-py3', type=('build', 'run'))
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/nvtop/package.py]: Can you provide some more info on these dependencies ? I think only the cuda driver which includes nvml is needed (not the cuda runtime, or these python wrappers)

**Review by baberlevi — DISMISSED** [review]

**Comment by perrymil** [comment]
removed the unnecessary py nvml and nvidia-ml package dependencies. Tested again with cuda as build dependency and it worked.

File: var/spack/repos/builtin/packages/nvtop/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    depends_on('cmake')
+    depends_on('ncurses')
+    depends_on('git')
+    depends_on('cuda', type=('build'))
+
+    def cmake_args(self):
+        cmake_args = []
+        cmake_args.append('-DNVML_RETRIEVE_HEADER_ONLINE=True')
+        return cmake_args
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/nvtop/package.py]:
```suggestion
        return [self.define('NVML_RETRIEVE_HEADER_ONLINE', True)]
```

File: var/spack/repos/builtin/packages/nvtop/package.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    homepage = "https://github.com/Syllo/nvtop"
+    url      = "https://github.com/Syllo/nvtop/archive/1.1.0.tar.gz"
+
+    version('1.1.0', sha256='00470cde8fc48d5a5ed7c96402607e474414d94b562b21189bdde1dbe6b1d1f3')
+
+    depends_on('cmake')
+    depends_on('ncurses')
+    depends_on('git')
+    depends_on('cuda', type=('build'))
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/nvtop/package.py]: You definitely cuda this for link/runtime as well -- it's not just a build-only dependency.  Remove the second argument.

**Review by sethrj — CHANGES_REQUESTED** [review]
Thanks for the contribution! One important change, one minor one.

File: var/spack/repos/builtin/packages/nvtop/package.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    """Nvtop stands for NVidia TOP, a (h)top like task monitor for NVIDIA GPUs.
+    It can handle multiple GPUs and print information about them
+    in a htop familiar way."""
+
+    homepage = "https://github.com/Syllo/nvtop"
+    url      = "https://github.com/Syllo/nvtop/archive/1.1.0.tar.gz"
+
+    version('1.1.0', sha256='00470cde8fc48d5a5ed7c96402607e474414d94b562b21189bdde1dbe6b1d1f3')
+
+    depends_on('cmake')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nvtop/package.py]: This is usually not needed:
```suggestion
    depends_on('cmake')
```
unless the application needs to link some CMake internal libraries.

File: var/spack/repos/builtin/packages/nvtop/package.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on('ncurses')
+    depends_on('git')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nvtop/package.py]: In which way does this package depend on `git`?

File: var/spack/repos/builtin/packages/nvtop/package.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on('cuda')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nvtop/package.py]: Would it make sense to inherit also from `CudaPackage` and drop this dependency?

**Review by alalazo — CHANGES_REQUESTED** [review]
See comments below

File: var/spack/repos/builtin/packages/nvtop/package.py (hunk 7 above)
Comment (perrymil) [reply var/spack/repos/builtin/packages/nvtop/package.py]: I added CudaPackage as a class and tried installing again. There was an error that I think has something to do with the cmake nvml part, but not quite sure what that means yet. 

`CMakeFiles/nvtop.dir/extract_gpuinfo.c.o: In function update_graphical_process':
/tmp/root/spack-stage/spack-stage-nvtop-1.1.0-u3qhvtr25atty7xujgix2aku3vhem6lz/spack-src/src/extract_gpuinfo.c:214: undefined reference to nvmlDeviceGetGraphicsRunningProcesses_v2'
CMakeFiles/nvtop.dir/extract_gpuinfo.c.o: In function update_compute_process':
/tmp/root/spack-stage/spack-stage-nvtop-1.1.0-u3qhvtr25atty7xujgix2aku3vhem6lz/spack-src/src/extract_gpuinfo.c:246: undefined reference to nvmlDeviceGetComputeRunningProcesses_v2'`

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Review by sethrj — APPROVED** [review]

**Pull request merged**